src/aggregate.d

```diff
@@ -639,6 +639,16 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
         return type;
     }
 
+    /****************************************
+     * Returns `true` if `this` aggregate has an `InputRange` interface
+     */
+    final bool isInputRange()
+    {
+        return (search(Loc(), Id.Ffront) &&
+                search(Loc(), Id.FpopFront) &&
+                search(Loc(), Id.Fempty));
+    }
+
```

Comment:
While a good idea to encapsulate this sort of thing, if this returns true the caller will have to call search() again to find it.